test/unit/concerns/host_managed_extensions_test.rb

```diff
@@ -0,0 +1,35 @@
+require 'test_plugin_helper'
+
+class HostManagedExtensionsTest < ActiveSupport::TestCase
+  before do
+    @role1 = FactoryGirl.create(:ansible_role)
+    @role2 = FactoryGirl.create(:ansible_role)
+    @role3 = FactoryGirl.create(:ansible_role)
+
+    @hostgroup_parent = FactoryGirl.create(:hostgroup, :ansible_roles => [@role2])
+    @hostgroup = FactoryGirl.create(:hostgroup, :parent => @hostgroup_parent)
```

Comment:
This is creating a lot of objects every time for no reason (they're not used everywhere)